avm/ptn/deployment-script/import-image-to-acr/main.bicep

```diff
@@ -0,0 +1,246 @@
+  name: managedIdentityName ?? 'id-ContainerRegistryImport'
+  location: location
+  tags: tags
+}
+
+// assign the Contributor role to the managed identity (new or existing) to import images into the ACR
+resource acrRoleAssignmentExistingManagedIdentities 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for i in range(0, length(assignRbacRole ? (managedIdentities.?userAssignedResourcesIds ?? []) : [])): if (useExistingManagedIdentity) {
+    name: guid('roleAssignment-acr-${existingManagedIdentities[i].name}')
+    scope: acr
+    properties: {
+      principalId: existingManagedIdentities[i].properties.principalId
+      roleDefinitionId: subscriptionResourceId(
+        'Microsoft.Authorization/roleDefinitions',
+        'b24988ac-6180-42a0-ab88-20f7382dd24c'
+      )
+      principalType: 'ServicePrincipal' // See https://docs.microsoft.com/azure/role-based-access-control/role-assignments-template#new-service-principal to understand why this property is included.
+    }
+  }
+]
+resource acrRoleAssignmentNewManagedIdentity 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (!useExistingManagedIdentity) {
+  name: guid('roleAssignment-acr-${newManagedIdentity.name}')
+  scope: acr
+  properties: {
+    principalId: newManagedIdentity.properties.principalId
+    roleDefinitionId: subscriptionResourceId(
+      'Microsoft.Authorization/roleDefinitions',
+      'b24988ac-6180-42a0-ab88-20f7382dd24c'
+    )
+    principalType: 'ServicePrincipal' // See https://docs.microsoft.com/azure/role-based-access-control/role-assignments-template#new-service-principal to understand why this property is included.
+  }
+}
+
+module imageImport 'br/public:avm/res/resources/deployment-script:0.2.3' = {
+  name: name ?? 'ACR-Import-${last(split(replace(image,':','-'),'/'))}'
+  scope: resourceGroup()
+  params: {
+    name: name
+    location: location
+    tags: tags
+    managedIdentities: managedIdentities
+    kind: 'AzureCLI'
+    runOnce: runOnce
+    azCliVersion: '2.59.0'
+    timeout: 'PT30M' // set timeout to 30m
+    retentionInterval: 'PT1H' // cleanup after 1h
+    environmentVariables: {
+      secureList: [
+        {
+          name: 'acrName'
+          value: acrName
+        }
+        {
+          name: 'imageName'
+          value: image
+        }
+        {
+          name: 'overwriteExistingImage'
+          value: toLower(string(overwriteExistingImage))
+        }
+        {
+          name: 'initialDelay'
+          value: '${string(initialScriptDelay)}s'
+        }
+        {
+          name: 'retryMax'
+          value: string(retryMax)
+        }
+        {
+          name: 'retrySleep'
+          value: '5s'
+        }
+      ]
+    }
+    cleanupPreference: cleanupPreference
+    storageAccountResourceId: storageAccountResourceId
+    containerGroupName: '${resourceGroup().name}-infrastructure'
+    subnetResourceIds: subnetResourceIds
+    // scriptContent: loadTextContent('./scripts/importscript.sh')
```

Comment:
Didn't work?

Reply:
No, it didn't. For now, I will leave it inline as it was.

Reply:
I can't say I like it - like at all 😄 , but give that we don't have a spec for that yet I'll comply. Just think it's a very bad practice maintainability-wise.